File: bubble/bubble.jl
function bubble_sort!(arr::Vector)
    n = length(arr)
    for i in 1:n
        swapped = false
        for j in 1:(n - i)
            if arr[j] > arr[j + 1]
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swapped = true
            end
        end
        !swapped && break
    end
    return arr
end

File: bubble/bubble.py
def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        swapped = False
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swapped = True
        if not swapped:
            break

def print_array(arr):
    for val in arr:
        print(val, end=" ")
    print()

if __name__ == "__main__":
    arr = [64, 34, 25, 12, 22, 11, 90]
    print("Original array: ", end="")
    print_array(arr)
    bubble_sort(arr)
    print("Sorted array: ", end="")
    print_array(arr)


